Source/GUI/MemViewer/MemViewer.cpp

```diff
@@ -190,13 +240,13 @@ bool MemViewer::handleNaviguationKey(const int key)
       m_byteSelectedPosY--;
       viewport()->update();
     }
-    else
+    else if (verticalScrollBar()->value() > verticalScrollBar()->minimum())
     {
       m_byteSelectedPosX = 15;
-      m_byteSelectedPosY = 15;
+      m_byteSelectedPosY = 0;
       scrollContentsBy(0, 1);
```

Comment:
The scroll should be BEFORE assigning `m_byteSelectedPosY` because if you set it THEN scroll, it's going to be on the last column of the same row instead of being the last column of the previous row.  Since you do it first for the right arrow key, do so here to remain consistent.

Reply:
Ah yeah, the change that makes the selection move when scrolling changed this behaviour. Actually this entire part should be refactored, it's not taking the variable number of rows and columns into account.